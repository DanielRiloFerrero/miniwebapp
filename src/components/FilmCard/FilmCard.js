import React, { useState } from 'react'

import {
  Col,
  Card, CardImg, CardBody,
  CardTitle, CardSubtitle
} from 'reactstrap';
import { format } from 'date-fns'

import FilmPopup from '../FilmPopup/FilmPopup'

import notAvailable from '../../_assets/images/not_available.svg'
import { IMG_URL } from '../../_assets/js/const'
import styles from './FilmCard.module.scss'

const FilmCard = props => {
  const [openPopup, setOpenPopup] = useState(false)

  const srcImg = props.data.poster_path ? `${IMG_URL}${props.data.poster_path}` : notAvailable
  const releaseDate = props.data.release_date ? format(new Date(props.data.release_date), 'dd/MM/yyyy') : 'Desconocida'

  const handleOnClick = () => setOpenPopup(true)
  const handleClosePopup = () => setOpenPopup(false)

  return (
    <>
      <Col sm='4' className={styles.card}>
        <Card onClick={() => handleOnClick()}>
          <CardImg top width="100%" src={srcImg} alt="Card image cap" />
          <CardBody>
            <CardTitle>{props.data.title}</CardTitle>
            <CardSubtitle>{releaseDate}</CardSubtitle>
          </CardBody>
        </Card>
      </Col>

      {openPopup && <FilmPopup data={props.data} close={() => handleClosePopup()} />}
    </>
  )
}

export default FilmCard