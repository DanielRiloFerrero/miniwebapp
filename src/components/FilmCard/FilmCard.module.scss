@import '../../_assets/styles/abstracts/vars.scss';

.card {
  padding: $p-xs 0;
}