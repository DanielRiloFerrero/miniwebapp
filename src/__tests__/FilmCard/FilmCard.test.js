import React from "react"
import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import FilmCard from "../../components/FilmCard/FilmCard"

const data = {
  'poster': 'dummy',
  'release_date': new Date()
}

describe('FilmCard tests', () => {
  it("render", () => {
    const { container } = render(<FilmCard data={data} />);
    expect(container).toMatchSnapshot()
  })
})