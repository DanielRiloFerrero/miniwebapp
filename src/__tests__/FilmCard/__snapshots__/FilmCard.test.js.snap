// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmCard tests render 1`] = `
<div>
  <div
    class="card col-sm-4"
  >
    <div
      class="card"
    >
      <img
        alt="Card image cap"
        class="card-img-top"
        src="not_available.svg"
        width="100%"
      />
      <div
        class="card-body"
      >
        <div
          class="card-title"
        />
        <div
          class="card-subtitle"
        >
          15/05/2020
        </div>
      </div>
    </div>
  </div>
</div>
`;
